charts/argo-services/templates/events/sync_notification_sensor.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Sensor
+metadata:
+  name: sync-notification
+spec:
+  template:
+    serviceAccountName: operate-workflow-sa
+  dependencies:
+  - name: sync-notification
+    eventSourceName: sync-notification
+    eventName: argocd
+  triggers:
+  - template:
+      name: trigger
+      argoWorkflow:
+        group: argoproj.io
+        version: v1alpha1
+        resource: workflows
+        operation: submit
+        namespace: {{ .Values.workflowsNamespace }}
+        parameters:
+          - dest: spec.arguments.parameters.0.value
+            src:
+              dependencyName: sync-notification
+              dataKey: body.application
+          - dest: spec.arguments.parameters.1.value
+            src:
+              dependencyName: sync-notification
+              dataKey: body.commitSha
+          - dest: spec.arguments.parameters.2.value
+            src:
+              dependencyName: sync-notification
+              dataKey: body.commitUrl
+          - dest: spec.arguments.parameters.3.value
+            src:
+              dependencyName: sync-notification
+              dataKey: body.argoUrl
+          - dest: spec.arguments.parameters.4.value
```

Comment:
Oof. It's not the most elegant CRD, this, is it? 😅 I couldn't see a better way either, though 🤷 I think it'll be ok since we're not forcing teams to work with this stuff day-to-day - should be pretty low-maintenance with any luck.